evaluation/model_evaluation.py

```
@@ -0,0 +1,1144 @@
+                            else:
+                                func(model_answer, correct_answer, user_input=prompt)
+
+                            async with results_lock:
+                                if func_name not in results["function_metrics"]:
+                                    results["function_metrics"][func_name] = {
+                                        "passed": 0,
+                                        "total": 0,
+                                    }
+                                results["function_metrics"][func_name]["passed"] += 1
+                                results["function_metrics"][func_name]["total"] += 1
+                        except (AssertionError, Exception) as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)